sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -372,4 +385,56 @@ class SidechainApp @Inject()
   def getSecretSubmitProvider: SecretSubmitProvider = secretSubmitProvider
 
   actorSystem.eventStream.publish(SidechainAppEvents.SidechainApplicationStart)
+
+  val timeoutDuration: FiniteDuration = settings.restApi.timeout
+
+  implicit val timeout: Timeout = Timeout(timeoutDuration)
+
+  /***
+   * Retrieve the SidechainBlockId needed to rollback the SidechainStateStorage for the backup.
+   * It's calculated by the following formula:
+   * Genesis_MC_block_height + (current_epch-2) * withdrawalEpochçength -1
+   * @param view
+   * @return sidechainBlockId to rollback or Exception if it's not found
+   */
+  def getSidechainBlockIdForBackUpRollback(view: View):Array[Byte] = {
+    val currentEpoch = view.state.getWithdrawalEpochInfo.epoch
+    val genesisMcBlockHeight = view.history.getMainchainCreationBlockHeight
+    val withdrawalEpochLength = sidechainSettings.genesisData.withdrawalEpochLength
+    val blockHeightToRollback = genesisMcBlockHeight + (currentEpoch -2) * withdrawalEpochLength - 1
+    val mainchainBlockReferenceInfo = view.history.getMainchainBlockReferenceInfoByMainchainBlockHeight(blockHeightToRollback).get()
+    mainchainBlockReferenceInfo.getMainchainReferenceDataSidechainBlockId
+  }
+
+  def createBackup(): Unit = {
```

Comment:
I think we need to keep the backup creation outside the SidechainApp. It should be another utility class that can be called separately and should avoid initialization the whole part needed for the sidechain app.
Method `getSidechainBlockIdForBackUpRollback` can be something called in the preparation phase, for example, using API. So the utility class will take the `sidechain block id` as a parameter.

Reply:
Done